packages/http/src/mocker/callback/callbacks.ts

```diff
@@ -9,12 +7,45 @@ import * as TaskEither from 'fp-ts/lib/TaskEither';
 import * as ReaderTaskEither from 'fp-ts/lib/ReaderTaskEither';
 import { head } from 'fp-ts/lib/Array';
 import { pipe } from 'fp-ts/lib/pipeable';
+
+import { resolveRuntimeExpressions } from '../../utils/runtimeExpression';
+import { IHttpOperationConfig, IHttpRequest, IHttpResponse } from '../../types';
 import { generate as generateHttpParam } from '../generator/HttpParamGenerator';
 import { validateOutput } from '../../validator';
 import { parseResponse } from '../../utils/parseResponse';
 import withLogger from '../../withLogger';
 import { violationLogger } from '../../utils/logger';
 import { Logger } from 'pino';
+import { RangeOrNumber } from '@stoplight/prism-core';
+
+export function scheduleCallback({
+  callback,
+  request,
+  response,
+  config,
+}: {
+  callback: IHttpCallbackOperation;
+  request: IHttpRequest;
+  response: IHttpResponse;
+  config: Pick<IHttpOperationConfig, 'callbackDelay' | 'callbackCount'>
+}) {
+  return withLogger(logger => {
+    let executionsLeft = reduceRange(config.callbackCount || 1);
+
+    const execute = () => {
+      if (executionsLeft > 0) {
+        setTimeout(() => {
+          executionsLeft--;
+
+          runCallback({ callback, request, response })(logger)()
```

Comment:
This might be a code smell. If a function is returning a Reader, we need to return it as it is and not execute it by using `withLogger`. We're likely going to need to refactor this.